lib/modules/manager/npm/post-update/yarn.ts

```diff
@@ -248,7 +248,9 @@ export async function generateLockFile(
         } else if (isYarnDedupeAvailable && s === 'highest') {
           commands.push(`yarn dedupe --strategy ${s}${cmdOptions}`);
         } else {
-          logger.debug(`yarn dedupe ${s} not available`);
+          commands.push(`yarn dlx yarn-berry-deduplicate --strategy ${s}`);
```

Comment:
no dynamic download, create a tool constraint instead

Reply:
FYI https://github.com/christophehurpeau/renovate/blob/17ba90f2fc3f61aade9f99a9be6300fb8692312f/lib/modules/manager/npm/post-update/yarn.ts#L244-L245

Reply:
How should I do that ?

Reply:
like the yarn command

Reply:
add the tool here like yarn

https://github.com/renovatebot/renovate/blob/de3cb95e91569768608df6a10d7e166f221c1243/lib/util/exec/containerbase.ts#L173-L177

add the tool to this array when needed

https://github.com/renovatebot/renovate/blob/de3cb95e91569768608df6a10d7e166f221c1243/lib/modules/manager/npm/post-update/yarn.ts#L101

Reply:
I'm not sure I understand how that would work. Would this is supposed to work with `yarn-deduplicate` also ? If so, maybe you could show how that would be refactored with that existing tool so I could make the same for this PR ?

Reply:
sure, but probably only next week

Reply:
@viceice is this still todo?